### PR TITLE
Api provides user to cancel request

### PR DESCRIPTION
**Pull request by pgaunitz**
### User Story
```
As a resident
In order to change my requests
I would like to be able to cancel unaccepted requests
```
## Changes proposed in this pull request:
* Adds destroy action to pongs controller, user can cancel their active pong
* Adds show action to pongs controller, user can see their last pong
* Adds serializer for show action
## What I have learned working on this feature:
* How to do a delete action
* That begin rescue is good stuff

### CODE REVIEW

File: app/controllers/pongs_controller.rb [hunk 1]
```diff
@@ -23,12 +24,35 @@ def update
...
+    render json: @pong, serializer: PongShowSerializer
+  rescue StandardError => e
+    render json: { message: 'You have not requested anything from a neighbour' }
+  end
+
   private
 
   def is_part_of_community?
     if current_user.community_status == 'accepted'
+
```
Comment (kfrostare) [inline app/controllers/pongs_controller.rb]: is this line break supposed to be here? If yes then ignore me.

File: spec/requests/user_can_cancel_request_spec.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+RSpec.describe 'DELETE /pongs/:id' do
+  let(:user) { create(:user, role: 'user', community_status: 'accepted') }
+  let(:user_credentials) { user.create_new_auth_token }
+  let(:user_headers) do
+    { HTTP_ACCEPT: 'application/json' }.merge!(user_credentials)
+  end
+  let(:ping) { create(:ping) }
+  
+  describe 'user can cancel their request' do
+    let!(:pong) { create(:pong, ping_id: ping.id, user_id: user.id, active: false, status: 'accepted') }
```
Comment (kfrostare) [inline spec/requests/user_can_cancel_request_spec.rb:12]: Is this Let supposed to be extra indented? If yes, ignore me.

File: spec/requests/user_can_cancel_request_spec.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    it 'destroys request' do
+      expect(User.find(user.id).pongs.last).not_to eq pong2.id
+    end
+
+    it 'returns success message' do
+      expect(response_json['message']).to eq 'Your request is removed'
+    end
+  end
+  describe 'user can cancel their request' do
```
Comment (kfrostare) [inline spec/requests/user_can_cancel_request_spec.rb:33]: Should there be a space in here between the end and the new describe block? If not, just ignore me.

**Review by kfrostare — APPROVED** [review]
Looking good //Kf

**Review by kaylawoodbury — APPROVED** [review]
Looks great! There are a few message responses that probably won't be used by the front end, but as we talked about I think they are good as a catch incase something goes wrong that the app doesn't break.

**Pull request merged**